awx/ui/test/e2e/tests/test-users-crud.js

```diff
@@ -42,6 +42,11 @@ const store = {
 
 module.exports = {
     before: (client, done) => {
+        // generate a unique username on each attempt.
+        const uniqueUser = uuid().substr(0, 8);
+        for (let i = 0; i < store.length; ++i) {
+            store[i].username += `-${uniqueUser}`;
```

Comment:
given enough retries, this will eventually result in unexpected validation errors from the api due to character length. Can you change it to replace the username each time instead of appending to it?